### PR TITLE
[Shop] Product show associations

### PR DESCRIPTION
**Pull request by mpysiak**
| Q               | A
|-----------------|-----
| Branch?         | bootstrap-shop
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml [hunk 1]
```diff
@@ -141,5 +141,19 @@
                 template="@SyliusShop/product/show/page/info/summary/variant_selection/add_to_cart.html.twig"
             />
         </service>
+
+        <service
+            id="sylius_shop.twig.component.product_association"
+            class="Sylius\Bundle\ShopBundle\Twig\Component\Product\ProductAssociationComponent"
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/Resources/config/services/twig/component.xml]:
```suggestion
            class="Sylius\Bundle\ShopBundle\Twig\Component\Product\AssociationComponent"
```

File: src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig [hunk 2]
```diff
@@ -1,8 +1,7 @@
 {% set product = hookable_metadata.context.product %}
 
 {% for association in product.associations %}
-    {{ render(url('sylius_shop_partial_product_association_show', {'productId': product.id, 'id': association.id, 'template': '@SyliusShop/product/show/page/product_listing/association.html.twig'})) }}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig:4]: This partial route could be removed

File: src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig [hunk 3]
```diff
@@ -1,8 +1,7 @@
...
+    {{ component('sylius_shop:product:association', { productAssociation: association }) }}
 {% else %}
-    <div>{{ 'sylius.ui.latest_products'|trans }}</div>
-    {{ render(url('sylius_shop_partial_product_index_latest', {'count': 4, 'template': '@SyliusShop/shared/product/horizontal_list.html.twig'})) }}
+    {{ component('sylius_shop:homepage:latest_products', { limit: 4 }) }}
```
Comment (GSadee) [inline src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig:6]: It should definitely change name and be moved to common directory

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig (hunk 2 above)
Comment (mpysiak) [reply src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig:4]: It is removed

File: src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig (hunk 3 above)
Comment (mpysiak) [reply src/Sylius/Bundle/ShopBundle/templates/product/show/page/product_listing/associations.html.twig:6]: It will be done in other PR

**Review by GSadee — APPROVED** [review]

**Pull request merged**